### PR TITLE
Document `import.meta.env.ASSETS_PREFIX` for CDN support

### PR DESCRIPTION
**Pull request by bluwy**
#### What kind of changes does this PR include?

- New or updated content

#### Description

Complements https://github.com/withastro/astro/pull/6714

RFC: https://github.com/withastro/roadmap/pull/545

This documents `import.meta.env.ASSETS_PREFIX` as added by the implementation PR. The `build.assetsPrefix` option is documented in the above PR's JSDoc.

Since CDN support isn't a common setup for most sites, I didn't document a new section for it. I think this could be enough as a start.

(CI linkcheck seems to fail because this PR relies on the auto-generated docs for the linked PR above)

### CODE REVIEW

**Review by sarah11918 — APPROVED** [review]
Approved once the feature is merged and the link check passes! Thanks @bluwy! :partying_face:

**Review by Yan-Thomas — APPROVED** [review]

**Pull request merged**
